pynamodb_attributes/timestamp.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime, timezone
+
+import pynamodb.constants
+from pynamodb.attributes import Attribute
+
+
+class TimestampAttribute(Attribute):
+    """"
+    Stores times as Unix epoch timestamps in a DynamoDB number.
+
+    >>> class MyModel(Model):
+    >>>   created_at_seconds = TimestampAttribute(default=lambda: datetime.now(tz=timezone.utc))
+    >>>   created_at_ms = TimestampMsAttribute(default=lambda: datetime.now(tz=timezone.utc))
+    """
+    attr_type = pynamodb.constants.NUMBER
+    _multiplier = 1.0
+
+    def deserialize(self, value: str) -> datetime:
+        return datetime.fromtimestamp(int(value) / self._multiplier, tz=timezone.utc)
+
+    def serialize(self, value: datetime) -> str:
+        return str(int(value.timestamp() * self._multiplier))
+
+    def __set__(self, instance, value: datetime) -> None:
+        if not isinstance(value, datetime):
+            raise TypeError(f"value has invalid type '{type(value)}; datetime expected")
+        if value.tzinfo is None:
```

Comment:
turns out that [not only does `tzinfo` need to be non-`None`, but `tzinfo.utcoffset` must also return non-`None`](https://stackoverflow.com/questions/5802108/how-to-check-if-a-datetime-object-is-localized-with-pytz).

Reply:
😮